File: Questions/102.Binary Tree Level Order/Solution.cpp
#include<iostream>
//#include<deque>
class TreeNode{
    public:
        int val;
        TreeNode* left;
        TreeNode* right;
        TreeNode(): val(val), left(left), right(right) {}
};

std::vector<std::vector<int>> levelOrder(TreeNode* root) {
    std::deque<TreeNode*> dq;
    std::vector<std::vector<int>> ans;
    if (!root) return ans;
    dq.push_back(root);
    while (!dq.empty()) {
        int size = dq.size();
        std::vector<int> curLevel;
        for(int i = 0; i < size; i++) {
            TreeNode* node = dq.front();
            curLevel.push_back(node->val);
            dq.pop_front();
            if(node->left) dq.push_back(node->left);
            if(node->right) dq.push_back(node->right);
        }
        ans.push_back(curLevel);
    }
    return ans;
}

File: Questions/102.Binary Tree Level Order/Solution.ts
class TreeNode{
    val: number
    left: TreeNode | null
    right: TreeNode | null
    constructor(val: number, left?: TreeNode | null, right?: TreeNode | null) {
        this.val = val
        this.left = (left === undefined ? null : left)
        this.right = (right === undefined ? null : right)
    }
}

function levelOrder(root: TreeNode | null): number[][] {
    if (!root) return []
    const dq: TreeNode[] = []
    const ans: number[][] = []
    dq.push(root)
    let head = 0;
    while (head < dq.length) {
        const curLevel: number[] = []
        const size = dq.length - head
        for(let j = 0; j < size; j++) {
            const node = dq[head++]
            curLevel.push(node.val)
            if (node.left) dq.push(node.left)
            if (node.right) dq.push(node.right)
        }
        ans.push(curLevel)
    }
    return ans
}